src/main/java/org/apache/commons/beanutils2/converters/DimensionConverter.java

```diff
@@ -0,0 +1,112 @@
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.commons.beanutils2.converters;
+
+import java.awt.Dimension;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * <p>
+ *     Allows converting configuration values to a {@link Dimension}.
+ *     This can be used to load settings for default dimensions of {@link java.awt.Component}s
+ *     {@link java.awt.Window}s, or {@link java.awt.Shape}s.
+ * </p>
+ *
+ * <p>
+ *     Accepts a single {@link Integer} value, or two {@link Integer} values
+ *     separated by the character <code>x</code>.
+ * </p>
+ *
+ * <p>The dimensions must not be negative, and must be {@link Integer} values.</p>
+ *
+ * @since 2.0.0
+ */
+public class DimensionConverter extends AbstractConverter {
+
+    /** Regular expression used to validate and tokenizer the {@link String}.  */
+    private static final Pattern DIMENSION_PATTERN = Pattern.compile("(?<x>\\d+)(?:x(?<y>\\d+))?");
+
+    /**
+     * Construct a <b>{@link Dimension}</b> <i>Converter</i> that throws
+     * a {@code ConversionException} if an error occurs.
+     */
+    public DimensionConverter() {
+        super();
+    }
+
+    /**
+     * Constructs a {@link org.apache.commons.beanutils2.Converter} that will return
+     * the specified default value if a conversion error occurs.
+     *
+     * @param defaultValue The default value to be returned
+     * if the value to be converted is missing or an error
+     * occurs converting the value.
+     */
+    public DimensionConverter(final Object defaultValue) {
+        super(defaultValue);
+    }
+
+    /**
+     * Gets the default type this {@code Converter} handles.
+     *
+     * @return The default type this {@code Converter} handles.
+     * @since 2.0.0
+     */
+    @Override
+    protected Class<?> getDefaultType() {
+        return Dimension.class;
+    }
+
+    /**
+     * @param value The String property value to convert.
+     * @return A {@link Dimension} which represents the configuration property value.
+     * @throws NullPointerException If the value is null.
+     * @throws NumberFormatException If the {@link Dimension} width or height is bigger than {@link Integer#MAX_VALUE}.
+     */
+    @Override
+    protected <T> T convertToType(Class<T> type, Object value) throws Throwable {
+        if (Dimension.class.isAssignableFrom(type)) {
+            Objects.requireNonNull(value, "Dimensions can not be null.");
+            final String stringValue = value.toString();
```

Comment:
Please rebase on master, then reuse the new convenience method in `AbstractConverter` to replace:
```
            Objects.requireNonNull(value, "Value must not be null.");
            final String stringValue = value.toString();
```
with:
```
final String stringValue = toString(value);
```